alien.go

```diff
@@ -449,6 +449,43 @@ func (m *Mux) Delete(path string, h func(http.ResponseWriter, *http.Request)) er
 	return m.AddRoute("DELETE", path, h)
 }
 
+// ContainsRoute checks if a route is present in the Mux.
+// It takes two arguments: path (string) and method (string).
+// If method is empty, it will look for the route in all HTTP methods.
+// It returns a boolean value indicating whether the route is found or not, along with an error if any.
+func (m *Mux) ContainsRoute(path, method string) (bool, error) {
+	findRoute := func(method string) (bool, error) {
+		_, err := m.find(method, path)
+		if err == nil {
+			return true, nil
+		}
+		if !errors.Is(err, errRouteNotFound) {
+			return false, err
+		}
+		return false, nil
+	}
+	if method != "" {
+		return findRoute(method)
+	}
+	methods := []string{
```

Comment:
Can we move this into global scope ? We don't need to allocate it in every `ContainsRoute` call.

Reply:
Yes, certainly. I've put it in the var statement:

```var (
	allMethods       = []string{"GET", "PUT", "POST", "HEAD", "PATCH", "OPTIONS", "CONNECT", "TRACE"}
	eof              = rune(0)
	errRouteNotFound = errors.New("route not found")
	errBadPattern    = errors.New("bad pattern")
	errUnknownMethod = errors.New("unkown http method")
	headerName       = "_alien"
)
```